### ISSUE TITLE
Flaky MeshAccessLog e2e test

### ISSUE THREAD

**Issue by michaelbeaumont** [kumahq/kuma#5104]
### What happened?

In https://github.com/kumahq/kuma/pull/5102 we marked the test as flaky. It's easy to reproduce with only one other suite in `e2e_env/universal` enabled e.g. `"Zone Egress"`. Config appears fine, it may be an issue with the `AccessLogStreamer`.

With https://github.com/kumahq/kuma/pull/5125 the test is no longer flaky but the issue still remains!

**Comment by jakubdyszkiewicz** [comment]
Is it still a problem?

**Comment by michaelbeaumont** [comment]
@jakubdyszkiewicz I think so https://github.com/kumahq/kuma/pull/5125#discussion_r992265683

**Comment by slonka** [comment]
@lobkovilya - you were the last ci rota, was it flakey still?

**Comment by slonka** [comment]
If I'm understanding this right we're even in a worse situation. The tests is not flaky but the behaviour is, right? I'm moving back to `triage` to talk about this but I would suggest renaming the issue and chasing down the flaky behaviour.

**Comment by michaelbeaumont** [comment]
Yes, something is apparently wrong, but not the behavior that was being tested. IIRC
